modules/file-transfer/src/main/java/org/apache/ignite/internal/network/file/FileTransferServiceImpl.java

```diff
@@ -0,0 +1,402 @@
+                new FileSender(
+                        nodeName,
+                        configuration.value().senderThreadPoolSize(),
+                        configuration.value().chunkSize(),
+                        new RateLimiter(configuration.value().maxConcurrentRequests()),
+                        (recipientConsistentId, message) -> messagingService.send(recipientConsistentId, FILE_TRANSFERRING_CHANNEL,
+                                message)
+                ),
+                new FileReceiver(nodeName, configuration.value().receiverThreadPoolSize())
+        );
+    }
+
+    @TestOnly
+    FileTransferServiceImpl(
+            TopologyService topologyService,
+            MessagingService messagingService,
+            Path tempDirectory,
+            FileSender fileSender,
+            FileReceiver fileReceiver
+    ) {
+        this.topologyService = topologyService;
+        this.messagingService = messagingService;
+        this.tempDirectory = tempDirectory;
+        this.fileSender = fileSender;
+        this.fileReceiver = fileReceiver;
+    }
+
+    @Override
+    public void start() {
+        topologyService.addEventHandler(new TopologyEventHandler() {
+            @Override
+            public void onDisappeared(ClusterNode member) {
+                fileReceiver.cancelTransfersFromSender(member.id());
+            }
+        });
+
+        messagingService.addMessageHandler(FileTransferMessageType.class,
+                (message, senderConsistentId, correlationId) -> {
+                    if (message instanceof FileDownloadRequest) {
+                        processDownloadRequest((FileDownloadRequest) message, senderConsistentId, correlationId);
+                    } else if (message instanceof FileUploadRequest) {
+                        processUploadRequest((FileUploadRequest) message, senderConsistentId, correlationId);
+                    } else if (message instanceof FileTransferErrorMessage) {
+                        processFileTransferErrorMessage((FileTransferErrorMessage) message);
+                    } else if (message instanceof FileTransferInfoMessage) {
+                        processFileTransferInfo((FileTransferInfoMessage) message);
+                    } else if (message instanceof FileHeaderMessage) {
+                        processFileHeader((FileHeaderMessage) message);
+                    } else if (message instanceof FileChunkMessage) {
+                        processFileChunk((FileChunkMessage) message);
+                    } else {
+                        LOG.error("Unexpected message received: {}", message);
+                    }
+                });
+    }
+
+    private void processFileTransferErrorMessage(FileTransferErrorMessage message) {
+        fileReceiver.receiveFileTransferErrorMessage(message);
+    }
+
+    private void processUploadRequest(FileUploadRequest message, String senderConsistentId, long correlationId) {
+        completedFuture(UUID.randomUUID())
```

Comment:
Once again I see this pattern. What's the reason? Writing
```
UUID transferId = UUID.randomUUID();
```
looks absolutely fine. Why would you need to pollute your code with extra paddings? To avoid writing the type explicitly?